scripts/backup-restore.sh

```diff
@@ -0,0 +1,401 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ -z "$CK8S_CONFIG_PATH" ]; then
+  log "ERROR: CK8S_CONFIG_PATH is not set."
+  log "Please set and export the variable before running the script."
+  exit 1
+fi
+
+usage() {
+  echo "Usage: $0 <action> <target>"
+  echo "Actions: backup | restore"
+  echo "Targets: opensearch | harbor | rclone | velero"
+  exit 1
+}
+
+# Sets a timestamp log before every message, also colors it purple to make it easier to distinct from rest of the text because why not
+PURPLE='\033[0;35m'
+NOCOLOR='\033[0m'
+log() {
+  echo -e "[${PURPLE}$(date +'%Y-%m-%d %H:%M:%S')${NOCOLOR}]: $1"
+}
+
+# allows the script to be run from anywhere
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+REPO_ROOT=$(realpath "$SCRIPT_DIR/..")
+CK8S_CMD="$REPO_ROOT/bin/ck8s"
+
+if [ "$#" -ne 2 ]; then
+  usage
+fi
+
+ACTION="$1"
+TARGET="$2"
+
+run_backup() {
+  case "$1" in
```

Comment:
NIT: To make this a bit clear what argument this is for, maybe we should add

```suggestion
  local backup_target="${1}"
  case "${backup_target}" in
```

Mainly for the end of the function on the log message
```bash
  log "Error: Invalid backup target '$1'."
```